### PR TITLE
Make it possible to introduce a different train limit for a corporation

### PR DESCRIPTION
**Pull request by perwestling**
Add a train_limit ability, with attribute increase, that can be added
to a corporation to set a permanent increase in the train limit.

### CODE REVIEW

File: lib/engine/phase.rb [hunk 1]
```diff
@@ -31,6 +31,10 @@ def current
       @phases[@index]
     end
 
+    def train_limit(entity)
```
Comment (tobymao) [inline lib/engine/phase.rb:34]: entity = nil

Comment (tobymao) [reply lib/engine/phase.rb:34]: and handle nil entity

File: lib/engine/phase.rb [hunk 2]
```diff
@@ -118,5 +122,12 @@ def next!
       @index += 1
       setup_phase!
     end
+
+    private
+
+    def train_limit_increase(entity)
+      ability = entity.abilities(:train_limit)
```
Comment (tobymao) [inline lib/engine/phase.rb:129]: entity.abilities(:train_limit) { |ability| return ability.increase } 
0

File: lib/engine/phase.rb (hunk 1 above)
Comment (tobymao) [reply lib/engine/phase.rb:34]: i see i guess you enforce it to exist, i guess it's ok

File: lib/engine/phase.rb (hunk 2 above)
Comment (perwestling) [reply lib/engine/phase.rb:129]: Done

File: lib/engine/phase.rb (hunk 1 above)
Comment (perwestling) [reply lib/engine/phase.rb:34]: I think it is fine to not have default value here, and instead enforce it to be specified. Did update all calls, and they all had access to a corporation.

**Comment by perwestling** [comment]
Compare #1742 

Is this a good way of adding special abilities for a corporation in a title? 

An alternative is to add a description ability without any functionality (so it shows up on the charter) and let the title implement the ability.

I do think it is good to, if possible, show special rules for an entity, in the game.

@tobymao Have pushed a fix for review comment.

**Comment by tobymao** [comment]
what do you mean add special abilities for a corporation? doesn't 1846 do this?

**Review by tobymao — APPROVED** [review]

**Pull request merged**